pkg/tool/kubectl.go

```diff
@@ -44,8 +47,65 @@ func (k Kubectl) DeleteNamespace(namespace string) {
 	time.Sleep(3 * time.Second)
 
 	if err := k.exec.RunProcess("kubectl", "get", "namespace", namespace); err != nil {
```

Comment:
Just thinking: Would it make sense to have a receiver method like `RunWithProxy` here which you could pass a function as argument?

Reply:
sounds good, will refactor

Reply:
i did, not sure if this is good, can you PTAL @unguiculus